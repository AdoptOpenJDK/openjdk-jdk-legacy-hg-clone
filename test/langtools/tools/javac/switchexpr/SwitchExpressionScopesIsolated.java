/*
 * @test /nodynamiccopyright/
 * @bug 8206986
 * @summary Verify that scopes in rule cases are isolated.
 * @compile/fail/ref=SwitchExpressionScopesIsolated.out -XDrawDiagnostics --enable-preview -source ${jdk.version} SwitchExpressionScopesIsolated.java
 */

public class SwitchExpressionScopesIsolated {

    private String scopesIsolated(int i) {
        return switch (i) {
            case 0 -> { String res = ""; break res; }
            case 1 -> { res = ""; break res; }
            default -> { res = ""; break res; }
        };
    }

}
