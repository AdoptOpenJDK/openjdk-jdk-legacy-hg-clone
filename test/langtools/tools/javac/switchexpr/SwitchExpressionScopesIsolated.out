SwitchExpressionScopesIsolated.java:13:25: compiler.err.cant.resolve.location: kindname.variable, res, , , (compiler.misc.location: kindname.class, SwitchExpressionScopesIsolated, null)
SwitchExpressionScopesIsolated.java:13:41: compiler.err.cant.resolve.location: kindname.variable, res, , , (compiler.misc.location: kindname.class, SwitchExpressionScopesIsolated, null)
SwitchExpressionScopesIsolated.java:14:26: compiler.err.cant.resolve.location: kindname.variable, res, , , (compiler.misc.location: kindname.class, SwitchExpressionScopesIsolated, null)
SwitchExpressionScopesIsolated.java:14:42: compiler.err.cant.resolve.location: kindname.variable, res, , , (compiler.misc.location: kindname.class, SwitchExpressionScopesIsolated, null)
- compiler.note.preview.filename: SwitchExpressionScopesIsolated.java
- compiler.note.preview.recompile
4 errors
